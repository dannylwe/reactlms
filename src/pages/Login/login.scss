$color-blue: rgb(56, 54, 214);
a {color: $color-blue}

body {
    background-image: url("../../utilities/images/lilypads.png");
}


.form-login {
    display: flex;
}
.form-login {
    align-items: center;
    justify-content: center;
    margin-top: 3.375rem;
}

.password-input {
    padding-top: 1.5625rem;
}

.login-title {
    display: flex
}

.login-title {
    align-items: center;
    justify-content: center;
    margin-top: 12.5rem;
}

.login-btn {
    width: 100%;
}

.signup-login {
    padding: 0.3125rem;
}
