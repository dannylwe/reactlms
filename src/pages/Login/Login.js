import React, { Component } from "react";
import { withRouter, Link } from "react-router-dom";
import "./login.scss";
import { connect } from "react-redux";
import { loginAuthentication } from "../../actions/loginActions";

export class Login extends Component {
  state = {
    username: "",
    password: ""
  };

  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  onSubmit = e => {
    e.preventDefault();
    const { history } = this.props;
    const { username, password } = this.state;
    const loginDetails = { username, password };
    const loginUrl =
      " https://challenge3andela.herokuapp.com/api/v1/auth/login";

    this.props.loginAuthentication(loginUrl, loginDetails, history);
  };

  render() {
    return (
      <>
        {/* welcome message */}
        <div className="login-title">
          <Link to="/login">
            <h1>Welcome to SendIT</h1>
          </Link>
        </div>
        <div className="form-login">
          <form onSubmit={this.onSubmit}>
            {/* username input */}
            <div className="form-group center-form">
              <label for="emailInput">Username</label>
              <input
                type="text"
                className="form-control"
                id="emailInput"
                aria-describedby="emailHelp"
                placeholder="Enter username"
                name="username"
                onChange={e => this.handleChange(e)}
              />
              {/* password input */}
              <div className="password-input">
                <label for="passwordInput">Password</label>
                <input
                  type="password"
                  className="form-control"
                  id="passwordInput"
                  placeholder="Password..."
                  name="password"
                  onChange={e => this.handleChange(e)}
                />
              </div>
            </div>
            <div class="signup-login">
              <span>
                <text>Not a member, </text>
                <Link to="/signup">Sign up</Link>
              </span>
            </div>
            {/* submit button */}
            <button
              type="submit"
              className="btn btn-primary login-btn"
              id="loginButton"
            >
              Login
            </button>
          </form>
        </div>
      </>
    );
  }
}

export default withRouter(
  connect(
    null,
    { loginAuthentication }
  )(Login)
);
