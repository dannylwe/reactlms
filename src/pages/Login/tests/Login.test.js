import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import { shallow, mount } from "enzyme";
import Login from "../Login";

describe("Login", () => {
  it("should render without failing", () => {
    const wrapper = renderer.create(
      <MemoryRouter>
        <Login />
      </MemoryRouter>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});

describe("Login", () => {
  const component = shallow(<Login />);
  it("handleChange function should be called", () => {
    const event = {
      target: { name: "" }
    };
    event.preventDefault = jest.fn();
    component.instance().setState({ username: "example", password: "123344" });
    component.instance().handleChange(event);
    component
      .find("input")
      .at(1)
      .simulate("change", { target: { name: "username", value: 3 } });
    const state = component.state();
    expect(state.username).toEqual(3);
    component.instance().onSubmit(event);
  });
});
