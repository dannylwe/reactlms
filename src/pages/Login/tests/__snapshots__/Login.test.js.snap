// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render without failing 1`] = `
Array [
  <div
    className="login-title"
  >
    <a
      href="/login"
    >
      <h1>
        Welcome to SendIt
      </h1>
    </a>
  </div>,
  <div
    className="form-login"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group center-form"
      >
        <label
          for="emailInput"
        >
          Username
        </label>
        <input
          aria-describedby="emailHelp"
          className="form-control"
          id="emailInput"
          name="username"
          onChange={[Function]}
          placeholder="Enter username"
          type="text"
        />
        <div
          className="password-input"
        >
          <label
            for="passwordInput"
          >
            Password
          </label>
          <input
            className="form-control"
            id="passwordInput"
            name="password"
            onChange={[Function]}
            placeholder="Password..."
            type="password"
          />
        </div>
      </div>
      <div
        class="signup-login"
      >
        <span>
          <text>
            Not a member, 
          </text>
          <a
            href="/signup"
          >
            Signup
          </a>
        </span>
      </div>
      <button
        className="btn btn-primary login-btn"
        id="loginButton"
        type="submit"
      >
        Login
      </button>
    </form>
  </div>,
]
`;
